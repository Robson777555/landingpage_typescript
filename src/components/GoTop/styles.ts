import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;
    box-shadow: 0 2px 5px rgba(0,0,0,0.2);

    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
      box-shadow: 0 4px 8px rgba(0,0,0,0.3);
    }

    &:focus {
      outline: 2px solid ${theme.colors.secondaryColor};
      outline-offset: 2px;
    }

    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;