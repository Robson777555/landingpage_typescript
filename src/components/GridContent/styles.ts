// src/components/GridContent/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const HtmlContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    margin: ${theme.spacings.xhuge} 0;
    text-align: left;

    & > * {
      margin-bottom: ${theme.spacings.medium};
    }

    h1, h2, h3, h4, h5, h6 {
      margin: ${theme.spacings.large} 0;
    }

    pre {
      display: inline-block;
      margin: 0 auto;
      background: rgba(0,0,0,0.1);
      padding: ${theme.spacings.small};
      border-radius: ${theme.spacings.xsmall};
      overflow-x: auto;
      max-width: 100%;
    }

    blockquote {
      border-left: 3px solid ${theme.colors.primaryColor};
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      color: ${theme.colors.mediumGray};
    }

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
  `}
`;