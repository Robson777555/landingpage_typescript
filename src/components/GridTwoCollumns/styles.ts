// src/components/GridTwoCollumns/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.large};
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      text-align: center;
      justify-content: center;
      gap: ${theme.spacings.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.medium};

    /* Estilos para mobile */
    @media ${theme.media.lteMedium} {
      align-items: center;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    padding: ${theme.spacings.medium};
    max-width: 400px;
    width: 100%;

    img, .next-image {
      max-width: 100%;
      height: auto;
      width: 100%;
      object-fit: contain;
    }

    @media ${theme.media.lteMedium} {
      padding: ${theme.spacings.large} 0;
      max-width: 300px;
    }
  `}
`;

export const Title = styled.h2`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
