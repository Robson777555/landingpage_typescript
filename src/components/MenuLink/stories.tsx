import { Meta, StoryObj } from "@storybook/react";
import { MenuLink } from "./MenuLink";

const meta: Meta<typeof MenuLink> = {
  title: "Components/MenuLink",
  component: MenuLink,
  argTypes: {
    children: {
      control: "text",
      defaultValue: "Menu Item",
    },
    link: {
      control: "text",
      defaultValue: "#section",
    },
    newTab: {
      control: "boolean",
      defaultValue: false,
    },
  },
  args: {
    children: "Menu Item",
    link: "#section",
    newTab: false,
  },
};

export default meta;

type Story = StoryObj<typeof MenuLink>;

export const InternalLink: Story = {
  args: {
    link: "#section",
  },
};

export const ExternalLink: Story = {
  args: {
    link: "https://example.com",
    newTab: true,
  },
};

export const ActiveLink: Story = {
  args: {
    link: "#active",
  },
  parameters: {
    nextjs: {
      navigation: {
        pathname: "#active",
      },
    },
  },
};
