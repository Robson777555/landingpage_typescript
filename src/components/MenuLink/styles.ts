import styled, { css } from 'styled-components';

interface ContainerProps {
  $isActive?: boolean;
}

export const Container = styled.a<ContainerProps>`
  ${({ theme, $isActive }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.fonts.sizes.small};
    padding: ${theme.spacings.xsmall};
    color: ${$isActive ? theme.colors.secondaryColor : theme.colors.primaryColor};
    position: relative;
    text-transform: uppercase;
    transition: all 0.3s ease;
    cursor: pointer;

    &::after {
      content: '';
      position: absolute;
      bottom: 0.5rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background-color: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }

    &:hover {
      color: ${theme.colors.secondaryColor};

      &::after {
        left: 25%;
        width: 50%;
      }
    }

    @media ${theme.media.lteMedium} {
      font-size: ${theme.fonts.sizes.medium};
      padding: ${theme.spacings.medium};
      text-align: center;
      min-height: 44px;
      width: 100%;
    }
  `}
`;