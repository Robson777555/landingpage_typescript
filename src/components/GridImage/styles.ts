// src/components/GridImage/styles.js
import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.large};
      justify-items: center;
    }
  `}
`;

export const GridElement = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  ${({ theme }) => css`
    border-radius: ${theme.spacings.small};
    overflow: hidden;
    box-shadow: 0 4px 8px rgba(0,0,0,0.1);
    transition: transform 300ms ease-in-out;
    width: 100%;
    max-width: 340px;
    margin: 0 auto;
    padding: ${theme.spacings.medium};

    &:hover {
      transform: translateY(-5px);
    }

    img, .next-image {
      max-width: 220px;
      width: 100%;
      height: auto;
      margin: 0 auto;
      display: block;
    }

    h2, h3, h4, h5, h6 {
      font-size: 2.4rem;
      font-weight: bold;
      text-align: center;
      white-space: normal;
      word-break: break-word;
      width: 100%;
      line-height: 1.2;
      margin-bottom: ${theme.spacings.small};
    }

    @media (max-width: 768px) {
      padding: ${theme.spacings.small};
      max-width: 98vw;
      min-width: 0;
      h2, h3, h4, h5, h6 {
        font-size: 2rem;
        font-weight: bold;
        text-align: center;
        white-space: normal;
        word-break: break-word;
        width: 100%;
        line-height: 1.2;
      }
    }
  `}
`;
