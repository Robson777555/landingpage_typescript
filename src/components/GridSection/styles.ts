import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
      text-align: center;
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.xlarge};
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    position: relative;
    padding: ${theme.spacings.medium};
    background: ${theme.colors.white};
    border-radius: ${theme.spacings.small};
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
    overflow: hidden;

    &:hover {
      transform: translateY(-5px);
      box-shadow: 0 4px 8px rgba(0, 0, 0, 0.15);
    }

    ${HeadingContainer} {
      position: relative;
      left: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: 2.4rem;
      text-align: center;
      word-break: break-word;
      white-space: normal;
      font-weight: bold;
      @media (max-width: 768px) {
        font-size: 2rem;
      }
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 4rem;
      top: -2rem;
      left: 0.5rem;
      transform: rotate(0deg);
      color: ${theme.colors.mediumGray};
      opacity: 0.3;
      z-index: 0;
      @media (max-width: 768px) {
        font-size: 2.5rem;
        top: -1.2rem;
        left: 0.5rem;
      }
    }

    ${TextComponent} {
      text-align: left;
      font-size: 1.2rem;
      @media (max-width: 768px) {
        font-size: 1rem;
      }
    }
  `}
`;

export const Title = styled.h3`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
