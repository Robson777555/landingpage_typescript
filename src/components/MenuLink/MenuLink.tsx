'use client';
import React, { MouseEvent } from 'react';
import Link from 'next/link';
import { usePathname } from 'next/navigation';
import * as Styled from './styles';

interface MenuLinkProps {
  children: React.ReactNode;
  link: string;
  newTab?: boolean;
  onClick?: (event: MouseEvent<HTMLAnchorElement>) => void;
}

export const MenuLink = ({ children, link, newTab = false, onClick }: MenuLinkProps) => {
  const pathname = usePathname();
  const isInternalLink = link.startsWith('#') || link.startsWith('/');
  const isActive = pathname === link;

  const handleClick = (e: MouseEvent<HTMLAnchorElement>) => {
    if (onClick) {
      e.preventDefault();
      onClick(e);

      if (link.startsWith('#')) {
        setTimeout(() => {
          const targetElement = document.querySelector(link);
          if (targetElement) {
            targetElement.scrollIntoView({
              behavior: 'smooth',
              block: 'start',
            });
          }
        }, 300);
      }
    }
  };

  if (isInternalLink) {
    return (
      <Link href={link} passHref legacyBehavior>
        <Styled.Container
          $isActive={isActive}
          target={newTab ? '_blank' : '_self'}
          rel={newTab ? 'noopener noreferrer' : undefined}
          onClick={handleClick}
        >
          {children}
        </Styled.Container>
      </Link>
    );
  }

  return (
    <Styled.Container
      href={link}
      target={newTab ? '_blank' : '_self'}
      rel={newTab ? 'noopener noreferrer' : undefined}
      onClick={handleClick}
    >
      {children}
    </Styled.Container>
  );
};