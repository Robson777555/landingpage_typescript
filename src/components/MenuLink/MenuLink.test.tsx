import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { MenuLink } from './MenuLink';
import { theme } from '../../styles/theme';
import { describe, it, expect, vi } from 'vitest';

describe('<MenuLink />', () => {
  const renderMenuLink = (props = {}) => {
    const defaultProps = {
      children: 'Link Text',
      link: '#section',
    };

    return render(
      <ThemeProvider theme={theme}>
        <MenuLink {...defaultProps} {...props} />
      </ThemeProvider>
    );
  };

  it('should render internal link correctly', () => {
    renderMenuLink();
    const link = screen.getByRole('link', { name: 'Link Text' });
    expect(link).toHaveAttribute('href', '#section');
  });

  it('should handle external links with newTab', () => {
    renderMenuLink({
      link: 'https://external.com',
      newTab: true,
    });

    const link = screen.getByRole('link');
    expect(link).toHaveAttribute('target', '_blank');
    expect(link).toHaveAttribute('rel', 'noopener noreferrer');
  });

  it('should call onClick when clicked', () => {
    const mockClick = vi.fn();
    renderMenuLink({ onClick: mockClick });

    const link = screen.getByRole('link');
    fireEvent.click(link);

    expect(mockClick).toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    const { container } = renderMenuLink();
    expect(container.firstChild).toMatchSnapshot();
  });
});