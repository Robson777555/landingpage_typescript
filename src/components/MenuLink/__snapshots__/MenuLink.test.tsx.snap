// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MenuLink /> > should match snapshot 1`] = `
<a
  class="sc-bXdtCk kDKUyx"
  href="#section"
  target="_self"
>
  Link Text
</a>
`;
